webservices/server-integration/src/main/java/org/jboss/as/webservices/config/ServerConfigImpl.java

```diff
@@ -194,7 +194,7 @@ public Integer getVirtualHostPort(String hostname, boolean securePort) {
         Host undertowHost = getUndertowHost(hostInfo);
         if (undertowHost != null && !undertowHost.getServer().getListeners().isEmpty()) {
             if (!securePort) {
-                return undertowHost.getServer().getListeners().get(0).getBinding().getValue().getPort();
+                return undertowHost.getServer().getListeners().get(0).getBinding().getValue().getAbsolutePort();
```

Comment:
This is dangerous. as first listener can be anything.
It will work with shipped config but beyond that is unpredictable.